src/components/GoogleMapContainer.ts

```diff
@@ -222,6 +237,61 @@ class GoogleMapContainer extends Component<GoogleMapContainerProps, { alertMessa
             ? getStaticMarkerUrl(image.enumImage as string, this.props.defaultMakerIcon)
             : "";
     }
+
+    public static validateProps(props: GoogleMapContainerProps): string {
+        let errorMessage = "";
+        if (props.onClickEvent === "callMicroflow" && !props.onClickMicroflow) {
+            errorMessage = "A 'Microflow' is required for 'Markers' 'Call a microflow'";
+        } else if (props.onClickEvent === "callNanoflow" && !props.onClickNanoflow.nanoflow) {
+            errorMessage = "A 'Nanoflow' is required for 'Markers' 'Call a nanoflow'";
+        } else if (props.onClickEvent === "showPage" && !props.page) {
+            errorMessage = "A 'Page' is required for 'Markers' 'Show a page'";
+        }
+        if (errorMessage) {
+            errorMessage = `Error in google maps configuration: ${errorMessage}`;
+        }
+
+        return errorMessage;
+    }
+
+    private executeOnClickAction = (data: LatLng) => {
+        const { latitudeAttribute, locationsEntity, longitudeAttribute, mxform,
+                onClickEvent, onClickMicroflow, onClickNanoflow, openPageAs, page } = this.props;
+        const latitude = data.lat ? Math.round(data.lat * 100000000) / 100000000 : 0 ;
+        const longitude = data.lng ? Math.round(data.lng * 100000000) / 100000000 : 0;
+        // Note: The precision in the databae is limited to 20 digits before and 8 digits after the decimal point
+        const context = new mendix.lib.MxContext();
+        mx.data.create({
+            entity: locationsEntity,
+            callback: object => {
+                object.set(latitudeAttribute, latitude);
+                object.set(longitudeAttribute, longitude);
+                if (onClickEvent === "callMicroflow" && onClickMicroflow) {
+                    mx.ui.action(onClickMicroflow, {
+                        origin: mxform,
+                        params: { applyto: "selection", guids: [ object.getGuid() ] },
+                        error: error => window.mx.ui.error(`Error executing on click microflow ${onClickMicroflow} : ${error.message}`)
+                    });
+                } else if (onClickEvent === "callNanoflow" && onClickNanoflow.nanoflow) {
+                    context.setContext(object.getEntity(), object.getGuid());
+                    window.mx.data.callNanoflow({
+                        nanoflow: onClickNanoflow,
+                        origin: mxform,
+                        context,
+                        error: error => window.mx.ui.error(`Error while executing the on change nanoflow: ${error.message}`)
+                    });
+                } else if (onClickEvent === "showPage" && page) {
+                    context.setContext(object.getEntity(), object.getGuid());
```

Comment:
Move this to the line after creating the context or in the new function after separating
`const context = new mendix.lib.MxContext();`
`context.setContext(object.getEntity(), object.getGuid());`